### PR TITLE
Patientmodel

### PR DESCRIPTION
**Pull request by alexiscordts**
Lets try this again....Refactored to database

### CODE REVIEW

File: .vs/VSWorkspaceState.json [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests",
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests\\ControllerTests",
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests\\Fakes",
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests\\IntegrationTests",
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests\\ServiceTests",
+    "\\C#Backend\\InpatientTherapySchedulingProgramTests\\Utilities"
+  ],
+  "SelectedNode": "\\C#Backend\\InpatientTherapySchedulingProgramTests\\IntegrationTests\\UserServiceControllerTests.cs",
+  "PreviewInSolutionExplorer": false
+}
```
Comment (NgoTimothy) [inline .vs/VSWorkspaceState.json:14]: Do we need to check in this file?

File: C#Backend/InpatientTherapySchedulingProgram/Controllers/PatientController.cs [hunk 2]
```diff
@@ -103,21 +96,21 @@ public async Task<ActionResult<Patient>> PostPatient(Patient patient)
...
+                }
+            }
+            catch (DbUpdateConcurrencyException) {
+                throw;
+            }
+            return Ok(patient);
         }
     }
-}
+}
```
Comment (NgoTimothy) [inline C#Backend/InpatientTherapySchedulingProgram/Controllers/PatientController.cs:116]: Need newline at end of file

File: C#Backend/InpatientTherapySchedulingProgram/Models/Patient.cs [hunk 3]
```diff
@@ -46,5 +46,48 @@ public Patient()
...
+            {
+                return false;
+            }
+
+            if (Object.ReferenceEquals(this, patient))
+            {
+                return true;
+            }
+
+            return this.PatientId == patient.PatientId && this.FirstName == patient.FirstName && this.MiddleName == patient.MiddleName &&
```
Comment (NgoTimothy) [inline C#Backend/InpatientTherapySchedulingProgram/Models/Patient.cs:68]: When comparing strings, I think we need to do string1.Equals(string2) to compare the values and not just the reference.

File: C#Backend/InpatientTherapySchedulingProgram/Services/PatientService.cs [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+            }
+            catch (DbUpdateConcurrencyException)
+            {
+                throw;
+            }
+
+            return patient;
+        }
+
+        private bool PatientExists(int pid)
```
Comment (NgoTimothy) [inline C#Backend/InpatientTherapySchedulingProgram/Services/PatientService.cs:104]: Can we turn this into an async method to get better concurrency?

**Review by NgoTimothy — CHANGES_REQUESTED** [review]

File: C#Backend/InpatientTherapySchedulingProgram/Controllers/PatientController.cs (hunk 2 above)
Comment (alexiscordts) [reply C#Backend/InpatientTherapySchedulingProgram/Controllers/PatientController.cs:116]: What do you mean??

**Review by NgoTimothy — APPROVED** [review]

**Review by LucasKnoll — APPROVED** [review]

**Pull request merged**